devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,599 @@
+/*	Copyright 2020 SmartThings
+*
+*	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+*	in compliance with the License. You may obtain a copy of the License at:
+*
+*		http://www.apache.org/licenses/LICENSE-2.0
+*
+*	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+*	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+*	for the specific language governing permissions and limitations under the License.
+*
+*	Inovelli Dimmer
+*
+*	Copyright 2020 SmartThings
+*
+*/
+metadata {
+	definition (name: "Inovelli Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.switch", runLocally: false, minHubCoreVersion: '000.017.0012', executeCommandsLocally: false, genericHandler: "Z-Wave", mcdSync: true) {
+		capability "Actuator"
+		capability "Configuration"
+		capability "Energy Meter"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Sensor"
+		capability "Switch"
+		capability "Switch Level"
+		capability "Power Meter"
+
+		fingerprint mfr:"031E", prod:"0001", model:"0001", deviceJoinName: "Inovelli Dimmer Switch", vid: "SmartThings-smartthings-Inovelli_Dimmer"//Inovelli Dimmer LZW31-SN
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.switch.on", backgroundColor:"#00a0dc", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.switch.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.switch.on", backgroundColor:"#00a0dc", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.switch.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+		}
+		valueTile("power", "device.power", width: 2, height: 2) {
+			state "default", label:'${currentValue} W'
+		}
+		valueTile("energy", "device.energy", width: 2, height: 2) {
+			state "default", label:'${currentValue} kWh'
+		}
+		standardTile("refresh", "device.switch", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+	}
+
+	main(["switch","power","energy"])
+	details(["switch", "power", "energy", "refresh"])
+
+	preferences {
+		// Preferences template begin
+		parameterMap.each {
+			input (
+				title: it.name,
+				description: it.description,
+				type: "paragraph",
+				element: "paragraph"
+			)
+
+			switch(it.type) {
```

Comment:
Squash this code a little bit, like we did in e.g. https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/34355/files#diff-e1d1dd249a0c2fc0fcb81dae9b328d9cR57 or https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/28554/files#diff-2f6cf148de496ce7346eb5802c13be8dR78